### PR TITLE
Change the dokka output format from the javadoc style to the native dokka style

### PR DESCRIPTION
**Pull request by evanchooly**
The dokka format has come a long way since it first started.  It's now quite clean and easy to read. This will be a more idiomatic experience for our kotlin users.

### CODE REVIEW

**Comment by evanchooly** [comment]
image

File: build-parent/pom.xml [hunk 1]
```diff
@@ -971,6 +971,24 @@
...
+                                    <goal>jar</goal>
+                                </goals>
+                                <configuration>
+                                    <classifier>javadoc</classifier>
+                                    <classesDirectory>${project.build.directory}/dokka</classesDirectory>
+                                    <skipIfEmpty>true</skipIfEmpty>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
```
Comment (geoand) [inline build-parent/pom.xml:991]: How come is this needed?

**Comment by geoand** [comment]
The `MicroProfile TCKs Tests` failures are not related to this and have been fixed on `main`.

The `Devtools Tests ` failures look related

File: build-parent/pom.xml (hunk 1 above)
Comment (evanchooly) [reply build-parent/pom.xml:991]: The `dokka` goal only generates the docs. we need the extra step to create and attach the jar.

**Comment by evanchooly** [comment]
> The `Devtools Tests ` failures look related

ah!  found a typo in the template text.  rebased while I was at it to try to get the MP test fixes.

File: build-parent/pom.xml (hunk 1 above)
Comment (geoand) [inline build-parent/pom.xml:991]: @famod, @gsmet do you know anything about the necessity of this?

Comment (famod) [reply build-parent/pom.xml:991]: AFAICS the `dokka` goal (unlike the `javadocJar` goal) is not creating a javadoc jar, which is compensated by this jar-plugin execution.

Looks like a limitation of the plugin: https://kotlin.github.io/dokka/1.7.10/user_guide/maven/usage/ (see goal descriptions and config options)

Comment (geoand) [reply build-parent/pom.xml:991]: Gotcha, thanks

**Review by geoand — APPROVED** [review]

**Pull request merged**